### ISSUE TITLE
isDevicePermissionGranted should return something else than false when it can't detect permissions

### ISSUE THREAD

**Issue by GregOriol** [jitsi/lib-jitsi-meet#1560]
The method `isDevicePermissionGranted` on `JitsiMeetJS.mediaDevices` currently returns `true` or `false` when permission is granted or not.

Some browsers don't support the required APIs (Safari...) and the method returns `false` all the time. So, if you planned to display a warning in case permission is not granted, it is always displayed on those browsers...

I'd like to suggest that the method should return a third value, `null`, when it can't detect the permissions.

**Comment by saghul** [comment]
Probably `undefined` makes more sense here.

**Comment by GregOriol** [comment]
This also applies to Firefox (!) which has the navigator.permissions API (that Safari doesn't have), but doesn't support `camera` or `microphone` as queries (as discussed here: https://stackoverflow.com/questions/53147944/firefox-permission-name-member-of-permissiondescriptor-camera-is-not-a-vali)

**Comment by saghul** [comment]
Since the support for the permissions API is so different across the beard maybe switching the strategy here would be a better bet.

**Comment by eddsaura** [comment]
Hi, I have similar problems. I need to know if we have the permissions or not in firefox, before launching the mediaDevices.enumerateDevices

**Comment by saghul** [comment]
Firefox will always prompt for permission.

**Comment by eddsaura** [comment]
> Firefox will always prompt for permission.

Yes, but if the user deny it unintentionally I want to check it before launching the actual function to get the devices

**Comment by saghul** [comment]
I'm afraid there is no way to know that in Firefox without trying. They don't implement the "camera" and "microphone" permissions in the Permissions API: https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API
